sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -724,6 +724,8 @@ def pcollection_count_monitoring_infos(self, tag_to_pcollection_id):
 
     if self.tagged_receivers:
       for tag, receiver in self.tagged_receivers.items():
+        if str(tag) not in tag_to_pcollection_id:
```

Comment:
It would be weird that accessing the dict adds the entry in this case but if this is the solution then it would be better we created a copy of the dict we pass from here:
https://github.com/apache/beam/blob/2434d496571aa46efda1a18d9ccc319815b712f8/sdks/python/apache_beam/runners/worker/bundle_processor.py#L1038

Reply:
you're  right, but that causes another error. I might be looking at the wrog thing entirely.

Reply:
We should be able to drop the if pcollection_id check below